scripts/pylib/twister/twisterlib/twister_path.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+# vim: set syntax=python ts=4 :
+#
+# Copyright (c) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+
+from pathlib import Path, PosixPath
+
+ZEPHYR_BASE = os.getenv("ZEPHYR_BASE")
+canonical_zephyr_base = os.path.realpath(ZEPHYR_BASE)
+
+
+class TPath(os.PathLike):
+    def __init__(self, path, *args):
+        self.path = Path(path)
+
+        for p in args:
+            self.path = self._joinpath(p)
+
+    def get_longpath(self):
+        normalised = os.fspath(self.path.resolve())
+
+        if isinstance(self.path, PosixPath):
+            return Path(normalised)
+
+        # On Windows, without this prefix, there is 260-character path length limit.
+        if not normalised.startswith('\\\\?\\'):
+            normalised = '\\\\?\\' + normalised
+        return Path(normalised)
```

Comment:
> Problem: Paths were deemed the main culprit. Windows has a limit of 260 character file paths by default.
> Problem: Windows and Linux have a limit of 260 character filepath element.

So you are basically trying to solve the same problem that `--short-build-path` (see https://github.com/zephyrproject-rtos/zephyr/pull/41930) already solves, but in a different way.

Does this actually work? Because Ninja and Zephyr SDK Windows toolchain binaries are not exactly long-path friendly, relying on the "ANSI version" of the I/O functions.

Reply:
I've had to make some fixes before answering you (Zephyr test failures). Currently, this change gives us a PASS on all Zephyr tests and all Unit Tests relevant for Windows (So without i.a. `jobserver` and `QEMUHandler`). I've tested that on a private CI.